lib/p2p/Pool.ts

```diff
@@ -183,6 +183,25 @@ class Pool extends EventEmitter {
    * Attempt to create an outbound connection to a node using its known listening addresses.
    */
   private tryConnectNode = async (node: NodeConnectionInfo, retryConnecting = false) => {
+    if (!this.connectWithLastAddress(node)) {
+      this.connectWithAdvertisedAddresses(node, retryConnecting);
```

Comment:
We need to `await` both of these calls since they are async functions, this the reason the build is currently failing.

Reply:
ah again overlooked, silly me thanks !